### DISCUSSION TITLE
Share Prisma types with frontend application?

### DISCUSSION THREAD

**Discussion by nikzanda** [prisma/prisma#14909] [Q&A] [answered]
Is it possible to share auto-generated TypeScript types with a front-end application?

**Comment by austincrim** [comment, marked as answer]
Hey @nikzanda 👋,

The specifics depend on your repo setup, but it is definitely possible to do this. It usually looks like this:
```ts
import type { User } from '@prisma/client'
```

**Reply by nikzanda** [reply]
It would be great if there was a tool (like `graphql-code-generator`) that generates a file with all models types. I think it might be better suited for developers for frontend projects that can't access the backend project code.

**Reply by austincrim** [reply]
All generated types can be found in `node_modules/.prisma/client/index.d.ts`. You could potentially share just that file if needed.

It would have to be updated every time your `schema.prisma` changes, though.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
